app/helpers/search_helper.rb

```diff
@@ -1,12 +1,12 @@
 module SearchHelper
-  def check_posts_permissions
-    (current_user&.is_moderator || current_user&.is_admin ? Post : Post.undeleted)
+  # @param user [User] user to check
+  def get_accessible_posts(user)
```

Comment:
Ruby convention is not to use `get_` or `set_` - a getter method is named after what it returns, and a setter uses `!` after the name. I'd suggest `accessible_posts_for(user)` here.

Reply:
Yeah, old non-Ruby habits, @ArtOfCode-, will do